src/main/webapp/js/user-page-loader.js

```diff
@@ -139,6 +143,42 @@ function addCancleButtonFunction(editButton, cancleButton, saveButton, messageDi
   messageDiv.appendChild(cancleButton);
 }
 
+/*adding functionality to save button and append it to message div*/
+function addSaveButtonFunction(editButton, cancleButton, saveButton, messageDiv, bodyDiv, bodyText, message) {
+  saveButton.innerHTML = "Save";
+  saveButton.onclick = function() {
+    editButton.style.visibility="visible";
+    const messageId = message.id;
+    const url = '/editMessage?user=' + parameterUsername+"&messageId="+messageId+"&messageText="+bodyText.value;
+    fetch(url)
+    .then((response) => {
+      return response.json();
+    })
+    .then((messages) => {
+      const messagesContainer = document.getElementById('message-container');
+      if (messages.length == 0) {
+        messagesContainer.innerHTML = '<p>This user has no posts yet.</p>';
+      } else {
+        messagesContainer.innerHTML = '';
+      }
+      messages.forEach((message) => {
+        const messageDiv = buildMessageDiv(message);
+        messagesContainer.appendChild(messageDiv);
```

Comment:
Move forEach into the else block. It doesn't do anything if message.length == 0, but it would be better for clarity.